uofm_maintenance.module

```diff
@@ -92,6 +98,31 @@ function uofm_maintenance_regen_jp2_worker($object) {
 }
 
 /**
+ * Regenerate FITS derivative from OBJ.
+ *
+ * @param string|AbstractObject $object
+ *   the string PID or a Fedora Object
+ */
+function uofm_maintenance_regen_fits_worker($object) {
+  $obj = FALSE;
+  if (is_string($object)) {
+    module_load_include('inc', 'islandora', 'includes/utilities');
+    if (islandora_is_valid_pid($object)) {
+      $obj = islandora_object_load($object);
+    }
+  }
+  if (is_a($object, 'AbstractObject')) {
+    $obj = $object;
+  }
+  if (isset($obj['OBJ'])) {
+    module_load_include('inc', 'islandora_fits', 'includes/derivatives');
+    drush_log("Regenerate FITS for " . $obj->id, "ok");
+    islandora_fits_create_techmd($obj, TRUE, array('source_dsid' => 'OBJ'));
```

Comment:
Your call, I have lots of things to do. But splitting these out into a bunch of submodules should be relatively easy.